### PR TITLE
Bug/cuketest for ratelimitat api

### PR DESCRIPTION
**Pull request by Soujanya2107**
Ratelimit feature test at api level for login using post
Checks for Response Header  " x-rate-limit-retry-after-seconds"

### CODE REVIEW

**Comment by Soujanya2107** [comment]
@radek1st Is this is fine to be implemented on signup page as well?

File: tests/javascript-cucumber/Features/step_definitions/steps.js [hunk 1]
```diff
@@ -333,6 +334,73 @@ module.exports = function () {
     this.Then(/^user is displayed with success message$/, async function () {
         await this.driver.wait(until.elementLocated(By.className('ant-alert-message'), 60 * 1000))
     });
+    
+    //ratelimit feature
+    this.Given(/^User accesses API endpoint to login$/,async function () {
```
Comment (radek1st) [inline tests/javascript-cucumber/Features/step_definitions/steps.js:339]: I don't understand why you have this step. It seems you are doing 20+ calls to the API already in the next one

Comment (radek1st) [reply tests/javascript-cucumber/Features/step_definitions/steps.js:339]: ah, the next step is still Swagger correct?
If yes, then do Swagger 20 times plus a failing one on our website to see the error

File: tests/javascript-cucumber/Features/step_definitions/steps.js [hunk 2]
```diff
@@ -333,6 +334,73 @@ module.exports = function () {
...
+        this.driver.findElement(By.xpath("//textarea[@class='body-param__text']")).sendKeys(Key.BACK_SPACE);
+        textArea.sendKeys(JSON.stringify(jsonData));
+        var i;
+        for(i=0;i<22;i++){
+        this.driver.findElement(By.className("btn execute")).click()
+        }
+        
+    });
+
+    this.Then(/^Error should be received saying retry after seconds$/,async function () {
```
Comment (radek1st) [inline tests/javascript-cucumber/Features/step_definitions/steps.js:398]: the error displayed should be checked or Relaypay website, (although Swagger will show a response too)

**Comment by radek1st** [comment]
> @radek1st Is this is fine to be implemented on Business signup page as well?

yes please

File: tests/javascript-cucumber/Features/step_definitions/steps.js (hunk 1 above)
Comment (Soujanya2107) [reply tests/javascript-cucumber/Features/step_definitions/steps.js:339]: @radek1st Radek on our website i dont see any error showing up on our website , no error is shown even after 20+ attempts, it gets freezed only at the backend, thats why I did the test on swagger

File: tests/javascript-cucumber/Features/step_definitions/steps.js (hunk 2 above)
Comment (Soujanya2107) [reply tests/javascript-cucumber/Features/step_definitions/steps.js:398]: Isn't implemented on front end yet , at the moment No error at front end , not even Error:429

File: tests/javascript-cucumber/Features/step_definitions/steps.js (hunk 1 above)
Comment (radek1st) [reply tests/javascript-cucumber/Features/step_definitions/steps.js:339]: that's fine, we just need to add this functionality on the frontend

Comment (radek1st) [reply tests/javascript-cucumber/Features/step_definitions/steps.js:339]: @sonu-sonam could you please display an error message on the frontend when getting rate limit error from the backed?

File: tests/javascript-cucumber/Features/ratelimit.feature [hunk 3]
```diff
@@ -0,0 +1,12 @@
+
+Feature: User exceeds 20 login/signup attempts in a minute
+
+    Scenario:Login Rate limit exceeded 
+        Given User accesses API endpoint to login
+        When  credentials are entered and login attempted over twenty times
+        Then Too many request error should be received
```
Comment (Soujanya2107) [inline tests/javascript-cucumber/Features/ratelimit.feature:7]: This needs to be implemented , for the time being it displays credentials invalid when account freezes, So until the implementation in front end its checking for error message @radek1st

File: tests/javascript-cucumber/Features/step_definitions/steps.js [hunk 4]
```diff
@@ -333,6 +334,79 @@ module.exports = function () {
...
+            "password":"Genesis1234*"
+        };
+        const tryout = await this.driver.wait(until.elementLocated(By.className("try-out")),allowedTimeout);
+        tryout.click();
+        const textArea = await this.driver.wait(until.elementLocated(By.xpath("//textarea[@class='body-param__text']")),allowedTimeout);
+        await this.driver.wait(until.elementIsVisible(textArea), 60 * 1000);
+        textArea.click();
+        this.driver.findElement(By.xpath("//textarea[@class='body-param__text']")).sendKeys(Key.BACK_SPACE);
+        textArea.sendKeys(JSON.stringify(jsonData));
+        for(i=0;i<22;i++){
```
Comment (Soujanya2107) [inline tests/javascript-cucumber/Features/step_definitions/steps.js:358]: 20 clicks  on API and 21st failing attempt from relaypay website

File: tests/javascript-cucumber/Features/step_definitions/steps.js (hunk 1 above)
Comment (sonu-sonam) [reply tests/javascript-cucumber/Features/step_definitions/steps.js:339]: @radek1st yes, I have set the error to be shown whatever comes from the backend. Asked Salman to throw 'Too many requests!' error message for status 429.

**Pull request merged**